app/models/journal_aggregator.rb

```diff
@@ -0,0 +1,92 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class JournalAggregator
+  MAX_TEMPORAL_DISTANCE = 300 # Maximum time between two journals in seconds
+
+  def self.merge_journals(journals_array)
+    aggregated_journals = journals_array.dup
+    advance = journals_array.count == 0
+    while advance
+      result = aggregated_journals.dup
+      aggregated_journals.combination(2).each do |journal_a, journal_b|
```

Comment:
I am not sure if I agree with your first point. To my understanding [combination(2)](http://ruby-doc.org/core-2.1.5/Array.html#method-i-combination) takes an array an creates a new one which contains every possible pairing from the elements from the former array. An since the order within the pairs should not make a difference I have to object to your first point.